src/installers/InstallRuleInstaller.ts

```diff
@@ -113,6 +113,30 @@ async function installSubDirNoFlatten(profile: ImmutableProfile, rule: ManagedRu
     }
 }
 
+function getBestFitRule(matchingRules: ManagedRule[], fileParts: string[]) {
+    return matchingRules.map(value => {
+        const ruleParts = value.route.split('/').reverse();
+        let numberOfMatches = 0;
+        for (let i = 0; i < fileParts.length; i++) {
+            if (fileParts[i] === undefined || ruleParts[i] === undefined) {
+                break;
+            }
+            if (fileParts[i] === ruleParts[i]) {
+                numberOfMatches++;
+            }
+        }
+        return {
+            rule: value,
+            count: numberOfMatches
+        };
+    }).reduce((previousValue, currentValue) => {
```

Comment:
Nitpick: is there a sane fallback value of type `{rule, count}` that could be given as a starting value for this `reduce`? I think reduce would currently throw if the `matchingRules` arg is an empty array. It's currently controlled at the call site that this function is not called with empty arrays, but there's no guarantees that someone won't inadvertently change it in the future.